src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1449,6 +1449,10 @@ int wm_vuldet_process_agent_vulnerabilities(sqlite3 *db, OSHash *cve_table, scan
                 w_strdup(pkg->version, report->version);
             }
             w_strdup(pkg->arch, report->arch);
+            if(pkg->type) {
+                os_free(report->type);
```

Comment:
It isn't needed to free `report->type` here since `report` is allocated a few lines above and `type` is not used until this block.

Reply:
You are right, fixed !